@here/harp-mapview/lib/Tile.ts

```diff
@@ -1647,6 +1704,51 @@ export class Tile implements CachedResource {
         return layoutStyle;
     }
 
+    private computeResourceInfo(): void {
+        let heapSize = 0;
+        let num3dObjects = 0;
+        let numTextElements = 0;
+        let numUserTextElements = 0;
+
+        const aggregatedObjSize = {
+            heapSize: 0,
+            gpuSize: 0
+        };
+
+        for (const object of this.objects) {
+            if (object.visible) {
+                num3dObjects++;
+            }
+            MapViewUtils.estimateObject3dSize(object, aggregatedObjSize);
+        }
+
+        for (const group of this.textElementGroups.groups) {
+            numTextElements += group[1].elements.length;
+        }
+        numUserTextElements = this.userTextElements.length;
+
+        // 216 was the shallow size of a single TextElement last time it has been checked, 312 bytes
+        // was the minimum retained size of a TextElement that was not being rendered. If a
+        // TextElement is actually rendered, the size may be _much_ bigger.
+        heapSize += (numTextElements + numUserTextElements) * 312;
```

Comment:
Except from the picking info, we can make a pretty good worst-case estimation of how much memory a ``TextElement`` consumes. Ping me if you need help with this.

Reply:
Decided to keep current estimation, as TextElement handling will be revised thoroughly in the future.